pkg/matrix/matrix_test.go

```diff
@@ -23,14 +23,16 @@ import (
 func Test_FanOut(t *testing.T) {
 	tests := []struct {
 		name             string
-		matrix           []v1beta1.Param
+		matrix           *v1beta1.Matrix
```

Comment:
can you add a test case for a matrix without params?

Reply:
I should have been a bit more clear-- can you add a test case where matrix.Params is a nil slice? I think this will lead to a nil pointer dereference, which is why I'm suggesting adding test coverage.

Reply:
@EmmaMunley has a PR focused on empty arrays in https://github.com/tektoncd/pipeline/pull/6140

Reply:
ah this is my bad, I was thinking nil params would cause a panic here but I confused myself-- it would be a panic if fanout accepted a pointer to a matrix and that pointer was nil. since fanout takes a matrix rather than a pointer to a matrix this won't happen. I think it would be reasonable to swap *v1beta1.Matrix to v1beta1.Matrix in this struct instead of getting a pointer and then dereferencing it but consider this comment non blocking.

Reply:
Okay this has been updated to use the matrix itself instead of a pointer.